core/shared/src/main/scala/scalaz/zio/Task.scala

```diff
@@ -0,0 +1,341 @@
+
+  /** See [[scalaz.zio.ZIO.interruptible]]
+   */
+  final def interruptible[A](zio: Task[A]): Task[A] =
+    ZIO.interruptible(zio)
+
+  /** See [[scalaz.zio.ZIO.interruptibleMask]]
+   */
+  final def interruptibleMask[A](k: ZIO.InterruptStatusRestore => Task[A]): Task[A] =
+    ZIO.interruptibleMask(k)
+
+  /** See [[scalaz.zio.ZIO.lock]]
+   */
+  final def lock[A](executor: Executor)(zio: Task[A]): Task[A] =
+    ZIO.lock(executor)(zio)
+
+  /** See [[scalaz.zio.ZIO.mergeAll]]
+   */
+  final def mergeAll[A, B](in: Iterable[Task[A]])(zero: B)(f: (B, A) => B): Task[B] =
+    ZIO.mergeAll(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.mergeAllPar]]
+   */
+  final def mergeAllPar[A, B](in: Iterable[Task[A]])(zero: B)(f: (B, A) => B): Task[B] =
+    ZIO.mergeAllPar(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.raceAll]]
+   */
+  final def raceAll[A](zio: Task[A], ios: Iterable[Task[A]]): Task[A] =
+    ZIO.raceAll(zio, ios)
+
+  /** See [[scalaz.zio.ZIO.reduceAll]]
+   */
+  final def reduceAll[A](a: Task[A], as: Iterable[Task[A]])(f: (A, A) => A): Task[A] =
+    ZIO.reduceAll(a, as)(f)
+
+  /** See [[scalaz.zio.ZIO.reduceAllPar]]
+   */
+  final def reduceAllPar[A](a: Task[A], as: Iterable[Task[A]])(f: (A, A) => A): Task[A] =
+    ZIO.reduceAllPar(a, as)(f)
+
+  /** See [[scalaz.zio.ZIO.require]]
+   */
+  final def require[A](error: Throwable): IO[Throwable, Option[A]] => IO[Throwable, A] =
+    ZIO.require(error)
+
+  /** See [[scalaz.zio.ZIO.reserve]]
+   */
+  def reserve[A, B](reservation: Task[Reservation[Any, Throwable, A]])(use: A => Task[B]): Task[B] =
+    ZIO.reserve(reservation)(use)
+
+  /** See [[scalaz.zio.ZIO.runtime]]
+   */
+  final def runtime: UIO[Runtime[Any]] = ZIO.runtime
+
+  /** See [[scalaz.zio.ZIO.succeed]]
+   */
+  final def succeed[A](a: A): UIO[A] = ZIO.succeed(a)
+
+  /** See [[scalaz.zio.ZIO.succeedLazy]]
+   */
+  final def succeedLazy[A](a: => A): UIO[A] = ZIO.succeedLazy(a)
+
+  /** See [[scalaz.zio.ZIO.supervised]]
+   */
+  final def supervised[A](zio: Task[A]): Task[A] =
+    ZIO.supervised(zio)
+
+  /** See [[scalaz.zio.ZIO.supervise]]
+   */
+  final def supervise[A](zio: Task[A]): Task[A] =
+    ZIO.supervise(zio)
+
+  /** See [[scalaz.zio.ZIO.superviseWith]]
+   */
+  final def superviseWith[A](zio: Task[A])(supervisor: IndexedSeq[Fiber[_, _]] => UIO[_]): Task[A] =
+    ZIO.superviseWith(zio)(supervisor)
+
+  /** See [[scalaz.zio.ZIO.suspend]]
+   */
+  final def suspend[A](io: => Task[A]): Task[A] =
+    ZIO.suspend(io)
+
+  /** See [[scalaz.zio.ZIO.uninterruptible]]
+   */
+  final def uninterruptible[A](zio: Task[A]): Task[A] =
+    ZIO.uninterruptible(zio)
+
+  /** See [[scalaz.zio.ZIO.uninterruptibleMask]]
+   */
+  final def uninterruptibleMask[A](k: ZIO.InterruptStatusRestore => Task[A]): Task[A] =
+    ZIO.uninterruptibleMask(k)
+
+  /** See [[scalaz.zio.ZIO.unsandbox]]
+   */
+  final def unsandbox[A](v: IO[Cause[Throwable], A]): Task[A] = ZIO.unsandbox(v)
```

Comment:
All the methods of the `Task` object should accept and return types compatible with `Task`. The type `IO[Cause[Throwable], A]` is not compatible with `Task`, because `Cause[Throwable]` is not a subtype of `Throwable`. So we should remove this method from the `Task` object.

Reply:
It looks like the old code got this wrong too!